packages/platform-server/src/transfer_state.ts

```diff
@@ -55,6 +83,13 @@ function serializeTransferStateFactory(doc: Document, appId: string, transferSto
       return;
     }
 
+    if (appIdsWithTransferStateSerialized) {
+      if (appIdsWithTransferStateSerialized.has(appId)) {
+        warnStateTransferHappened(appId);
+      }
+      appIdsWithTransferStateSerialized.add(appId);
+    }
```

Comment:
We can move this logic into the `warnIfStateTransferHappened` function itself (I've also added `If` into the function name) and remove the `ngDevMode` check from the  `warnIfStateTransferHappened` function itself (it would be tree-shaken-away anyways).
```suggestion
    if (typeof ngDevMode === 'undefined' || ngDevMode) {
      warnIfStateTransferHappened(appId, injector);
    }
```